### PR TITLE
RFC: dynamically adjust column width at runtime for specific fields

### PR DESCRIPTION
**Pull request by cgzones**
Adjust the column width for fields known to be large dynamically at each cycle to the longest value.

Also with the recent addition of multiple screens, over-long columns can be moved into their own screen.

### CODE REVIEW

File: linux/LinuxProcess.c [hunk 1]
```diff
@@ -280,7 +280,9 @@ static void LinuxProcess_writeField(const Process* this, RichString* str, Proces
       }
       xSnprintf(buffer, n, "%5lu ", lp->ctxt_diff);
       break;
-   case SECATTR: snprintf(buffer, n, "%-30.30s ", lp->secattr ? lp->secattr : "?"); break;
+   case SECATTR:
+      snprintf(buffer, n, "%-*.*s ", Process_fieldWidths[SECATTR], Process_fieldWidths[SECATTR], lp->secattr ? lp->secattr : "N/A");
+      break;
```
Comment (BenBE) [inline linux/LinuxProcess.c:285]: If you keep CGROUP and CCGROUP in one line you might as well keep SECATTR in one line too …

File: linux/LinuxProcessList.c [hunk 2]
```diff
@@ -900,13 +900,17 @@ static void LinuxProcessList_readCGroupFile(LinuxProcess* process, openat_arg_t
 
    bool changed = !process->cgroup || !String_eq(process->cgroup, output);
 
+   Process_updateFieldWidth(CGROUP, strlen(output));
    free_and_xStrdup(&process->cgroup, output);
 
-   if (!changed)
+   if (!changed) {
+      Process_updateFieldWidth(CCGROUP, strlen(process->cgroup_short));
```
Comment (BenBE) [inline linux/LinuxProcessList.c:907]: `CCGROUP` given by `process->cgroup_short` can be `NULL` in which case this is aliased to `process->cgroup`.

File: linux/LinuxProcessList.c [hunk 3]
```diff
@@ -900,13 +900,17 @@ static void LinuxProcessList_readCGroupFile(LinuxProcess* process, openat_arg_t
...
       return;
+   }
 
    char* cgroup_short = CGroup_filterName(process->cgroup);
    if (cgroup_short) {
+      Process_updateFieldWidth(CCGROUP, strlen(cgroup_short));
       free_and_xStrdup(&process->cgroup_short, cgroup_short);
       free(cgroup_short);
    } else {
```
Comment (BenBE) [inline linux/LinuxProcessList.c:916]: Missing call to `Process_updateFieldWidth` in the `else` branch …

**Review by BenBE — CHANGES_REQUESTED** [review]

**Comment by BenBE** [comment]
Should also adjust CPU usage column for values above 10000% as of #947 …

**Comment by fasterit** [comment]
LGTM

**Pull request merged**